### DISCUSSION TITLE
Elastic Horovod Usage Question

### DISCUSSION THREAD

**Discussion by czmrand** [horovod/horovod#2781] [Q&A]
As far as I can tell, elastic horovod does not support recovery from a failure of the "dispatcher", i.e. the worker on which "horovodrun" is called. Can it?
1. If so, would it make sense to launch the "dispatcher" as a "non-preemptible" device?
2. If so, it would be nice if the dispatcher could be configured to be a relatively cheap machine, i.e. without a gpu.
3. What are the instructions for installing horovod with gloo on a non-gpu device? 
Right now I run:
pip uninstall --yes horovod
export HOROVOD_WITH_GLOO=1
pip install horovod
but get the error:
    nvcc fatal   : Unsupported gpu architecture 'compute_30'
    CMake Error at compatible_horovod_cuda_kernels_generated_cuda_kernels.cu.o.RelWithDebInfo.cmake:220 (message):

Thanks

**Comment by czmrand** [comment]
Turns out that the error I was getting was environment related...
Now I am able to install horovod with gloo, but am getting an error due to the fact that name of the NIC on the CPU instance (dispatcher) is different that the name on all of the GPU instances (workers).
I am able to work around this by hacking the code (horovod/runner/gloo_run.py)... but it would be nice if support for this possibility could be built-in.

**Comment by tgaddair** [comment]
Hey @czmrand, yes we normally run the driver on a dedicated CPU instance and the workers on preemptible GPU instances for the reason you describe.

Regarding the NIC issue, we have a PR in progress to address the issue (#2482), but had some concerns about the design.  Please let us know if you have some thoughts on this, and contributions welcome!

One way we commonly run the elastic mode to simplify the setup of driver and workers is to use [Ray](https://github.com/horovod/horovod/blob/master/docs/ray.rst#elastic-ray-executor).
